### ISSUE TITLE
[commitment] move verification to its own module

### ISSUE THREAD

**Issue by mimoo** [o1-labs/proof-systems#385]
A polynomial commitment is three algorithms:

1. commit
2. open (or as I like to call it, create an evaluation proof)
3. verify

These three algorithms are all specified in the same file which makes it hard to parse. We should move the verify stuff in its own module under the same crate. Specifically, [verify](https://github.com/o1-labs/proof-systems/blob/master/poly-commitment/src/commitment.rs#L946) function should be moved to `verify.rs` or something.

**Comment by sassafrasass** [comment]
will take a look at this
